gofakes3.go

```diff
@@ -184,7 +184,7 @@ func (g *GoFakeS3) httpError(w http.ResponseWriter, r *http.Request, err error)
 }
 
 func (g *GoFakeS3) listBuckets(w http.ResponseWriter, r *http.Request) error {
-	buckets, err := g.storage.ListBuckets()
+	buckets, err := g.storage.ListBuckets(r.Context())
```

Comment:
@ncw I refactored it to use the context. Could you have a look again? Then I can move on to fixing tests and other usage

Reply:
UPDATE: tests are fixed